### PR TITLE
Unify google-oauth2 access

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
Add a new `AuthGoogleAccessForm` component for google login/sign up.

## WHY
Having a single source of truth for which route to take makes our code more legible.

## HOW
Our omniauth configuration has two access [types](https://github.com/empirical-org/Empirical-Core/pull/9395/files#diff-73b9e84ca74d4d010f73cc756cb0557167455ca12d3eb3d64eceb9ecc5246f80R14) for google_oauth2, online and offline.  Each configuration has corresponding endpoints `/auth/google/online_access` and `/auth/google/offline_access` which were previously labeled `authentication_only` and `authentication_and_authorization` respectively.  Upon closer examination, I realized that `authentication_only` actually did authorization too, it just lacked the offline access, hence the renaming.

The distinction between online and offline access is tied to [refresh tokens](https://developers.google.com/identity/protocols/oauth2/openid-connect#refresh-tokens) but the overview is that the online access mode is preferred if our system already has a valid refresh token for a user.  If a user's refresh token is missing or expired, then the offline access flow is utilized to get a new one.

The one new piece of UI/UX is the `redirect to new_session_path(google_offline_access_expired: true)`.  Basically, if the check for a user's offline access fails, they are redirected to a sessions/new page which asks for re-authorization (see screenshot below).  @petersharkeyquill approved the copy and design for this new transition page.

### Screenshots
Screen Shot 2022-07-21 at 1 45 50 PM

### Notion Card Links
https://www.notion.so/quill/Fix-Google-oauth2-issue-c7615d7078c94c848317ca339bd19934

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES.
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | N/A

### CODE REVIEW

**Review by anathomical — APPROVED** [review]
Looks good to me.

File: services/QuillLMS/app/controllers/auth/google_controller.rb [hunk 1]
```diff
@@ -37,11 +40,11 @@ def authentication
...
   end
 
-  private def verify_authorization_for_offline_access
-    user = User.where('google_id = ? OR email = ?', @profile.google_id&.to_s, @profile.email&.downcase).first
+  private def check_for_authorization
+    user = User.find_by('google_id = ? OR email = ?', @profile.google_id&.to_s, @profile.email&.downcase)
     return if user.nil?
 
-    redirect_to GoogleIntegration::AUTHORIZATION_AND_AUTHENTICATION_PATH unless user.google_authorized?
+    redirect_to new_session_path(google_offline_access_expired: true) unless user.google_authorized?
```
Comment (dandrabik) [inline services/QuillLMS/app/controllers/auth/google_controller.rb]: Consider making this `return if user.google_authorized?` above

**Review by dandrabik — APPROVED** [review]
:+1: LGTM. Let's do a good amount of smoke testing, then 🚢 .

Thanks for putting this in place.

File: services/QuillLMS/app/controllers/auth/google_controller.rb (hunk 1 above)
Comment (brendanshean) [reply services/QuillLMS/app/controllers/auth/google_controller.rb]: Good idea

File: services/QuillLMS/client/app/bundles/Teacher/components/accounts/AuthGoogleAccessForm.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export const AUTH_GOOGLE_ONLINE_ACCESS_PATH = "/auth/google/online_access"
+export const AUTH_GOOGLE_OFFLINE_ACCESS_PATH = "/auth/google/offline_access"
+
+const AuthGoogleAccessForm = ({
+  buttonClass = '',
+  formClass = '',
+  offlineAccess = false,
+  showIcon = true,
+  spanClass = '',
+  text
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Teacher/components/accounts/AuthGoogleAccessForm.tsx:14]: Nice use of default parameters for default props.

**Review by emilia-friedberg — APPROVED** [review]
This looks good to me. I like your use of default props for that shared component.

File: services/QuillLMS/client/app/bundles/Teacher/components/accounts/AuthGoogleAccessForm.tsx (hunk 2 above)
Comment (brendanshean) [reply services/QuillLMS/client/app/bundles/Teacher/components/accounts/AuthGoogleAccessForm.tsx:14]: @emilia-friedberg I meant to ask you about this.  I chose empty string but wondered if `null` makes more sense here?

**Comment by brendanshean** [comment]
FFR, smoke testing: https://www.notion.so/quill/Fix-Google-oauth2-issue-c7615d7078c94c848317ca339bd19934

**Pull request merged**